File: leetcode/src/main/java/leetcode/p0101_0200/BestTimeToBuyAndSellStockII.java
package leetcode.p0101_0200;

/**
 * <a href="https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/">122. Best Time to Buy
 * and Sell Stock II</a>
 */
public class BestTimeToBuyAndSellStockII {
  /** @implNote Time {@code O(n)}, space {@code O(1)}. */
  public int maxProfit(int[] prices) {
    int result = 0;
    for (int i = 1; i < prices.length; i++) {
      if (prices[i] - prices[i - 1] > 0) {
        result += prices[i] - prices[i - 1];
      }
    }
    return result;
  }
}

File: leetcode-kt/src/main/kotlin/com/giwankim/leetcode/BestTimeToBuyAndSellStockII.kt
package com.giwankim.leetcode

class BestTimeToBuyAndSellStockII {
    fun maxProfit(prices: IntArray): Int {
        var result = 0
        for (i in 1 until prices.size) {
            val diff = prices[i] - prices[i - 1]
            if (diff > 0) {
                result += diff
            }
        }
        return result
    }
}
